django/contrib/admin/sites.py

```diff
@@ -280,6 +288,10 @@ def wrapper(*args, **kwargs):
             urlpatterns += [
                 re_path(regex, wrap(self.app_index), name='app_list'),
             ]
+
+        if self.final_catch_all_view:
+            urlpatterns.append(re_path(r'^.*$', wrap(self.catch_all_view)))
```

Comment:
Maybe
```suggestion
            urlpatterns.append(re_path(r'^(?P<url>.*)$', wrap(self.catch_all_view)))
```
and we could use `url` in the `catch_all_view()`:
```python
     def catch_all_view(self, request, url): 
         if (settings.APPEND_SLASH and self.append_slash and not url.endswith('/')):
            urlconf = getattr(request, 'urlconf', None)
            path = '%s/' % request.path
            ...
```

Reply:
OK, I pulled this in.